op-acceptance-tests/tests/flashblocks/utils.go

```diff
@@ -51,3 +56,42 @@ func (f *Flashblock) UnmarshalJSON(data []byte) error {
 
 	return nil
 }
+
+// listenForFlashblocks reads flashblocks from the given websocket client for the
+// specified duration. This mirrors the devstack DSL helper but lives inside the
+// test package so acceptance tests can reuse the logic.
+func listenForFlashblocks(logger log.Logger, wsClient *opclient.WSClient, duration time.Duration, output chan<- []byte, done chan<- struct{}) error {
+	defer close(done)
+
+	logger.Info("Listening on WebSocket client", "duration", duration)
+
+	timeout := time.After(duration)
+	messageCount := 0
+	for {
+		select {
```

Comment:
What guarantees this will have fully exited before the test completes? If the test failed so it didn't actually wait for `doneListening` to be closed this would still be running and potentially interfere with later tests. Even logging after the test completes causes errors as that shouldn't happen.